### PR TITLE
feat: add internal plugin metrics

### PR DESCRIPTION
**Pull request by swiatekm-sumo**
The plugin can now emit metrics describing its internal state. These describe the flow of data through the plugin in terms of the number of records and batches that were either sent or dropped, as well as buffer sizes. The metrics are disabled by default, and can be enabled by adding `EnableInternalMetrics = true` to the configuration.

### CODE REVIEW

File: test/test_metrics_writer.py [hunk 1]
```diff
@@ -1,7 +1,13 @@
 # -*- coding: utf-8 -*-
+# pylint: disable=redefined-outer-name,no-self-use
```
Comment (pmalek-sumo) [inline test/test_metrics_writer.py:2]: Why do we want to add those? Wouldn't it be better to improve the code and remove this exclusion rules?

File: test/test_metrics_sender.py [hunk 2]
```diff
@@ -35,6 +35,25 @@ def test_post_normal_no_additional_header():
     met_sender.cancel_timer()
 
 
+def test_sent_stats():
```
Comment (pmalek-sumo) [inline test/test_metrics_sender.py:38]: Is this used anywhere?

**Review by pmalek-sumo — COMMENTED** [review]
🤷

File: test/test_metrics_sender.py (hunk 2 above)
Comment (swiatekm-sumo) [reply test/test_metrics_sender.py:38]: It's a test, the test runner runs it.

File: test/test_metrics_writer.py (hunk 1 above)
Comment (swiatekm-sumo) [reply test/test_metrics_writer.py:2]: `redefined-outer-name` is annoying to make work with pytest's fixtures. Basically, the following code:

```
@pytest.fixture
def some_fixture():
  ...

def test_some_feature(some_fixture):
  do_something_with(some_fixture)
```
will cause pytest to call the test function with the computed value of `some_fixture()`, based on the parameter name. This isn't an issue at all - in fact, it's what makes pytest have good test setup ergonomics. But pylint just sees a function argument named the same as a global, and complains. In general, this is a valid complaint, just not here, which is why I'm disabling the warning only for this file.

**Review by pmalek-sumo — APPROVED** [review]
I'm not sure but 👍 I'd prefer someone else to take a look at this :)

**Review by kkujawa-sumo — APPROVED** [review]

**Pull request merged**